### DISCUSSION TITLE
Can json.dump() throw an exception?

### DISCUSSION THREAD

**Discussion by cc636489** [nlohmann/json#2832] [Q&A] [answered]
Hi there,

I'd like to know whether `json.dump()` function can throw an exception or not, so that I can make a decision to use a try catch block around it.

Thanks in advance for your attention!

**Comment by nlohmann** [comment, marked as answer]
Yes, the function can throw in case of invalid UTF-8 stored in a string, see https://json.nlohmann.me/api/basic_json/dump/, including the example.

**Comment by cc636489** [comment]
Thanks for the quite response! I appreciate it.


On Mon, Jun 21, 2021 at 1:46 PM Niels Lohmann ***@***.***>
wrote:

> Yes, the function can throw in case of invalid UTF-8 stored in a string,
> see https://json.nlohmann.me/api/basic_json/dump/, including the example.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nlohmann/json/discussions/2832#discussioncomment-899512>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFQJQACC3AQ462GBPCAG7BDTT6CKFANCNFSM47B4ZQWQ>
> .
>
-- 
Chen CHEN
Graduate Student in ICES&EM
POB 3.402J
The University of Texas at Austin

Phone: +1 512-968-2126
Email : ***@***.***
